### PR TITLE
[SPARK-49864][SQL][FOLLOW-UP] Fix default suggestion for binary arithmetic overflow

### PR DESCRIPTION
**Pull request by mihailomilosevic2001**
### What changes were proposed in this pull request?
Improvement on default branch for try suggestion.

### Why are the changes needed?
When we hit default branch in codeGen, we need to return a default value that would specify that we do not know the function, and not just a blank string.

### Does this PR introduce _any_ user-facing change?
Yes.

### How was this patch tested?
No branch hits this behaviour so far, but we need to safeguard from the possible errors.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala [hunk 1]
```diff
@@ -298,7 +298,7 @@ abstract class BinaryArithmetic extends BinaryOperator
           case "+" => "try_add"
           case "-" => "try_subtract"
           case "*" => "try_multiply"
-          case _ => ""
+          case _ => "unknown_function"
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala:301]: In which cases can't we suggest anything? For which functions are there no `try_*`?

Comment (mihailom-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala:301]: For now there is no case, but it seemed weird to throw another error within error parsing of the initial error and hide the actual cause of the error.

Comment (MaxGekk) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala:301]: I see, lgtm.

**Review by MaxGekk — APPROVED** [review]

**Comment by mihailom-db** [comment]
Let's wait for @srielau to review and then merge.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @mihailom-db.

**Review by srielau — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @mihailom-db and @srielau for review.

**Pull request closed without merging**